gum/backend-darwin/gummoduleregistry-darwin.c

```diff
@@ -130,29 +130,22 @@ gum_module_registry_on_image_removed (const struct mach_header * mh,
 }
 
 static void
-gum_module_registry_on_dyld_notification (GumInvocationContext * context,
-                                          gpointer user_data)
+gum_lldb_image_notifier (enum dyld_image_mode mode,
+                         guint32 info_count,
+                         const struct dyld_image_info info[])
 {
-  enum dyld_image_mode mode;
-  uint32_t info_count;
-  const struct dyld_image_info * info;
   uint32_t i;
 
-  mode =
-      GPOINTER_TO_SIZE (gum_invocation_context_get_nth_argument (context, 0));
-  if (mode != dyld_image_adding && mode != dyld_image_removing)
-    return;
-  info_count =
-      GPOINTER_TO_UINT (gum_invocation_context_get_nth_argument (context, 1));
-  info = gum_invocation_context_get_nth_argument (context, 2);
-
   for (i = 0; i != info_count; i++)
   {
     if (mode == dyld_image_adding)
       gum_add_image (info[i].imageLoadAddress, NULL);
     else
       gum_remove_image (info[i].imageLoadAddress);
   }
+
+  if (gum_dyld_notifier_context != NULL)
```

Comment:
Is this check needed? I guess this is racy regardless, since we're not holding the loader lock.

Reply:
let's remove it for simplicity

Reply:
turn into an assertion, so hopefully if we loose a race we'd know

Reply:
nah, racy is racy, removed the assertion too

Reply:
Sounds good 👍